### PR TITLE
Changed failed_at field as ISODate

### PR DESCRIPTION
**Pull request by BehroozBvk**
Fix #2605

Changed `failed_at` field to `ISODate`, When `config/queue.php` **failed.driver** is **mongodb**.

### CODE REVIEW

File: src/Queue/Failed/MongoFailedJobProvider.php [hunk 1]
```diff
@@ -28,7 +28,7 @@ public function log($connection, $queue, $payload, $exception)
             'connection' => $connection,
             'queue' => $queue,
             'payload' => $payload,
-            'failed_at' => Carbon::now()->getTimestamp(),
+            'failed_at' => new \MongoDB\BSON\UTCDateTime(now()),
```
Comment (alcaeus) [inline src/Queue/Failed/MongoFailedJobProvider.php:31]: Noted that `Carbon::now()` allows for mocking time in tests, so I presume that `now()` does the same, so we shouldn't rely on creating a new `UTCDateTime` instance without a time given.

**Comment by alcaeus** [comment]
@BehroozBvk there are some coding standards issues. You can fix these by running `./vendor/bin/phpcbf src/Queue/Failed/MongoFailedJobProvider.php`.

**Comment by GromNaN** [comment]
You can add a test in `QueueTest` with something like this:
```php
        // Log a failed job
        Carbon::setTestNow('2019-01-01 00:00:00');
        $provider->log('test_connection', 'test_queue', 'test_payload', new Exception('test_exception'));

        $failedJob = Queue::getDatabase()->table(Config::get('queue.failed.table'))->first();

        $this->assertSame('test_connection', $failedJob['connection']);
        $this->assertSame('test_queue', $failedJob['queue']);
        $this->assertSame('test_payload', $failedJob['payload']);
        $this->assertEquals(new UTCDateTime(now()), $failedJob['failed_at']);
        $this->assertStringStartsWith('Exception: test_exception in ', $failedJob['exception']);
```

**Comment by BehroozBvk** [comment]
@alcaeus Do you have any special settings for **phpcbf**? Because I used this command and it automatically corrected the codes.

**Comment by GromNaN** [comment]
> @alcaeus Do you have any special settings for **phpcbf**? Because I used this command and it automatically corrected the codes.
> 
> ```shell
> $ ./vendor/bin/phpcbf packages/laravel-mongodb/tests/QueueTest.php
> ```

The command must run from inside the `laravel-mongodb` directory to take the [`phpcs.xml`](https://github.com/mongodb/laravel-mongodb/blob/master/phpcs.xml.dist) config.

File: src/Queue/Failed/MongoFailedJobProvider.php [hunk 2]
```diff
@@ -24,13 +25,15 @@ class MongoFailedJobProvider extends DatabaseFailedJobProvider
...
     public function log($connection, $queue, $payload, $exception)
     {
-        $this->getTable()->insert([
-            'connection' => $connection,
-            'queue' => $queue,
-            'payload' => $payload,
-            'failed_at' => Carbon::now()->getTimestamp(),
-            'exception' => (string) $exception,
-        ]);
+        $this->getTable()->insert(
```
[the remark this answers is not in the document]
Comment (BehroozBvk) [reply src/Queue/Failed/MongoFailedJobProvider.php:28]: I used **phpcs.xml.dist** in the package exactly this time. I hope everything is ok.

```bash
./vendor/bin/phpcbf --standard=phpcs.xml.dist src/Queue/Failed/MongoFailedJobProvider.php 
```
```bash
./vendor/bin/phpcbf --standard=phpcs.xml.dist tests/QueueTest.php
```

**Review by GromNaN — APPROVED** [review]
Thank you. I reverted some style changes.

**Pull request merged**